packages/x-data-grid/src/models/gridBaseSlots.ts

```diff
@@ -67,6 +67,42 @@ export type SkeletonProps = {
   height?: number | string;
 };
 
+// InputComponent
+// variant="outlined"
+// size="small"
+// focusElementRef={valueRef}
+// disabled={readOnly}
+
+export type TextFieldProps = {
+  autoComplete?: string;
+  className?: string;
+  color?: 'primary' | 'error';
+  disabled?: boolean;
+  error?: boolean;
+  fullWidth?: boolean;
+  helperText?: string | null;
+  id?: string;
+  inputRef?: React.Ref<HTMLInputElement>;
+  label?: React.ReactNode;
+  onChange?: React.ChangeEventHandler;
+  onKeyDown?: React.KeyboardEventHandler;
+  placeholder?: string;
+  size?: 'small' | 'medium';
+  slotProps?: {
+    input?: {
+      disabled?: boolean;
+      endAdornment?: React.ReactNode;
+      startAdornment?: React.ReactNode;
+    };
+    htmlInput?: React.InputHTMLAttributes<HTMLInputElement>;
+  };
+  style?: React.CSSProperties;
+  tabIndex?: number;
+  type?: React.HTMLInputTypeAttribute;
+  value?: string;
+  variant?: 'standard' | 'outlined';
```

Comment:
@KenanYusuf With your recent design changes, do you think we can get rid of `variant` and use only the `'outlined'` one?

Reply:
Yeah, the variant names are also pretty Material UI specific. It's the sort of prop that will vary from one library to another.